### PR TITLE
support Variable Operation

### PR DESCRIPTION
**Pull request by duhanmin**
variable | result
-- | --
${yyyy-01-01} | 2021-01-01
${yyyy-01-01%-2y} | 2019-01-01
${yyyy-MM-01%-2M} | 2021-02-01
${yyyy-MM-dd%-2d} | 2021-03-31
${yyyy MM ----- HH%-1H} | 2021 04 ----- 14
${yyyyMMdd%-1d} | 20210401
${yyyyMM01%-1M} | 20210301
${HH%-1H} | 14

### CODE REVIEW

**Comment by duhanmin** [comment]
https://github.com/apache/incubator-linkis/issues/2416

File: linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/utils/VariableOperationUtils.java [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+                        buffer.replace(startIndex - 1, endIndex + 1, newContent);
+                    }else {
+                        buffer.replace(startIndex, endIndex + 1, newContent);
+                    }
+                    startIndex = buffer.indexOf(PLACEHOLDER_LEFT, startIndex + newContent.length());
+                }catch (IllegalArgumentException e1){
+                    startIndex = buffer.indexOf(PLACEHOLDER_LEFT, endIndex);
+
+                } catch (Exception e2){
+                    throw new RuntimeException(e2);
```
Comment (peacewong) [inline linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/utils/VariableOperationUtils.java:127]: Linkis exception should be thrown, you can refer to https://linkis.apache.org/zh-CN/community/development_specification/exception_catch

Comment (duhanmin) [reply linkis-commons/linkis-common/src/main/java/org/apache/linkis/common/utils/VariableOperationUtils.java:127]: 这里可以不抛异的,我改下

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**